metapkg/targets/generic/__init__.py

```diff
@@ -54,6 +55,14 @@ def __init__(self):
     def name(self):
         return f"Generic POSIX"
 
+    def prepare(self):
+        if platform.system() == "Darwin":
```

Comment:
Hmm. I'm not in love with how this composes.  Let's add a `MacOSGenericTarget` that shares `prepare()` with the regular `MacOSTarget` (we already have `LinuxGenericTarget`)

Reply:
Yeah, that sounds cleaner to me.